ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -0,0 +1,175 @@
+    }
+  }
+
+  /**
+   * Returns the target account address for the given name
+   * @param name ANS name
+   * @returns Account address | null
+   */
+  async getAddressByName(name: string): Promise<string | null> {
+    const { domain, subdomain } = name.match(namePattern)?.groups ?? {};
+    if (!domain) return null;
+    if (subdomain) return this.getAddressBySubdomainName(domain, subdomain);
+    return this.getAddressByDomainName(domain);
+  }
+
+  /**
+   * Returns the account address for the given domain name
+   * @param domain domain name
+   * @example
+   * if name is `aptos.apt`
+   * domain = aptos
+   *
+   * @returns account address | null
+   */
+  private async getAddressByDomainName(domain: string) {
+    if (domain.match(nameComponentPattern) === null) return null;
+    const ansResource: { type: Gen.MoveStructTag; data: any } = await this.provider.getAccountResource(
+      this.contractAddress,
+      `${this.contractAddress}::domains::NameRegistryV1`,
+    );
+    const data = ansResource.data as NameRegistryV1;
+    const { handle } = data.registry;
+    const domainsTableItemRequest = {
+      key_type: `${this.contractAddress}::domains::NameRecordKeyV1`,
+      value_type: `${this.contractAddress}::domains::NameRecordV1`,
+      key: {
+        subdomain_name: { vec: [] },
+        domain_name: domain,
+      },
+    };
+
+    try {
+      const item = await this.provider.getTableItem(handle, domainsTableItemRequest);
+      return item.target_address.vec[0];
+    } catch (error: any) {
+      // if item not found, response is 404 error - meaning item not found
+      if (error.status === 404) {
+        return null;
+      }
+      throw new Error(error);
+    }
+  }
+
+  /**
+   * Returns the account address for the given subdomain_name
+   * @param domain domain name
+   * @param subdomain subdomain name
+   * @example
+   * if name is `dev.aptos.apt`
+   * domain = aptos
+   * subdomain = dev
+   *
+   * @returns account address | null
+   */
+  private async getAddressBySubdomainName(domain: string, subdomain: string): Promise<string | null> {
+    if (domain.match(nameComponentPattern) === null) return null;
+    if (subdomain.match(nameComponentPattern) === null) return null;
+    const ansResource: { type: Gen.MoveStructTag; data: any } = await this.provider.getAccountResource(
+      this.contractAddress,
+      `${this.contractAddress}::domains::NameRegistryV1`,
+    );
+    const data = ansResource.data as NameRegistryV1;
+    const { handle } = data.registry;
+    const domainsTableItemRequest = {
+      key_type: `${this.contractAddress}::domains::NameRecordKeyV1`,
+      value_type: `${this.contractAddress}::domains::NameRecordV1`,
+      key: {
+        subdomain_name: { vec: [subdomain] },
+        domain_name: domain,
+      },
+    };
+
+    try {
+      const item = await this.provider.getTableItem(handle, domainsTableItemRequest);
+      return item.target_address.vec[0];
```

Comment:
Can we add explicit handling for when the vec is empty / has more than 1 item? Rather than just throwing some kind of "index out of range" error? Same with the other functions.

Reply:
if the `vec` is empty - it would just return null, and if it has more than 1 items - it would return the first item.